core/src/col_metadata.rs

```diff
@@ -67,39 +68,59 @@ impl MongoColMetadata {
             display_size: bson_type_info.fixed_bytes_length,
```

Comment:
@EthanHardyMongo I think you need to update this line to get the display size to behave has expected.
```suggestion
            display_size: bson_type_info.fixed_bytes_length(type_mode),
```